kirby/modules.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Kirby;
+
+use A;
+use F;
+use Dir;
+use Response;
+
+class Modules {
+
+  // core
+  public $controllers = array();
+  public $languages   = array();
+  public $models      = array();
+  public $snippets    = array();
+  public $tags        = array();
+  public $templates   = array();
+  public $autocss     = array();
+  public $autojs      = array();
+
+  // panel
+  public $blueprints  = array();
+  public $fields      = array();
+
+
+  /**
+   * Magic getter for module arrays
+   */
+  public function __call($method, $arguments) {
+    return isset($this->{$method}) ? $this->{$method} : null;
+  }
+
+  /**
+   * Registers a plugin directory for a module type
+   */
+  public function register($module, $dir) {
+    if(!isset($this->{$module})) return false;
+
+    // prepare for auto-assets
+    if($module === 'autocss' or $module === 'autojs') {
+      $route = 'assets/modules/' . sha1($dir);
+      $dir   = array($dir, $route);
```

Comment:
What does that do? Isn't `$dir`/`$root` supposed to be a string value in `assets()`?